libpod/adapter/runtime_remote.go

```diff
@@ -2,27 +2,174 @@
+		Conn:    conn,
 	}, nil
 }
 
 // Shutdown is a bogus wrapper for compat with the libpod runtime
 func (r RemoteRuntime) Shutdown(force bool) error {
 	return nil
 }
+
+// ContainerImage
+type ContainerImage struct {
+	remoteImage
+}
+
+type remoteImage struct {
+	ID          string
+	Labels      map[string]string
+	RepoTags    []string
+	RepoDigests []string
+	Parent      string
+	Size        int64
+	Tag         string
+	Repository  string
+	Created     time.Time
+	InputName   string
+	Names       []string
+	Digest      digest.Digest
+	isParent    bool
+}
+
+// GetImages returns a slice of containerimages over a varlink connection
+func (r *LocalRuntime) GetImages() ([]*ContainerImage, error) {
+	var newImages []*ContainerImage
+	images, err := iopodman.ListImages().Call(r.Conn)
+	if err != nil {
+		return nil, err
+	}
+	for _, i := range images {
+		name := i.Id
+		if len(i.RepoTags) > 1 {
+			name = i.RepoTags[0]
+		}
+		newImage, err := imageInListToContainerImage(i, name)
+		if err != nil {
+			return nil, err
+		}
+		newImages = append(newImages, newImage)
+	}
+	return newImages, nil
+}
+
+func imageInListToContainerImage(i iopodman.ImageInList, name string) (*ContainerImage, error) {
+	imageParts, err := image.DecomposeString(name)
+	if err != nil {
+		return nil, err
+	}
+	created, err := splitStringDate(i.Created)
+	if err != nil {
+		return nil, err
+	}
+	ri := remoteImage{
+		InputName:   name,
+		ID:          i.Id,
+		Labels:      i.Labels,
+		RepoTags:    i.RepoTags,
+		RepoDigests: i.RepoTags,
+		Parent:      i.ParentId,
+		Size:        i.Size,
+		Created:     created,
+		Tag:         imageParts.Tag,
+		Repository:  imageParts.Registry,
+		Names:       i.RepoTags,
+		isParent:    i.IsParent,
+	}
+	return &ContainerImage{ri}, nil
+}
+
+// NewImageFromLocal returns a container image representation of a image over varlink
+func (r *LocalRuntime) NewImageFromLocal(name string) (*ContainerImage, error) {
+	img, err := iopodman.GetImage().Call(r.Conn, name)
+	if err != nil {
+		return nil, err
+	}
+	return imageInListToContainerImage(img, name)
+
+}
+
+func splitStringDate(d string) (time.Time, error) {
+	fields := strings.Fields(d)
+	t := fmt.Sprintf("%sT%sZ", fields[0], fields[1])
+	return time.ParseInLocation(time.RFC3339Nano, t, time.UTC)
+}
+
+// IsParent goes through the layers in the store and checks if i.TopLayer is
+// the parent of any other layer in store. Double check that image with that
+// layer exists as well.
+func (ci *ContainerImage) IsParent() (bool, error) {
+	return ci.remoteImage.isParent, nil
+}
+
+// ID returns the image ID as a string
+func (ci *ContainerImage) ID() string {
+	return ci.remoteImage.ID
+}
+
+// Names returns a string array of names associated with the image
+func (ci *ContainerImage) Names() []string {
+	return ci.remoteImage.Names
+}
+
+// Created returns the time the image was created
+func (ci *ContainerImage) Created() time.Time {
+	return ci.remoteImage.Created
+}
+
+// Size...
```

Comment:
I think we can be more verbose than that :^)

Reply:
this code is going to be deprecated and removed very very soon